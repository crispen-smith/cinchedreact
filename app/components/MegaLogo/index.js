/**
 *
 * MegaLogo
 *
 */

import React from 'react';
import styled from 'styled-components';

const MegaLogoBase = styled.div`
  flex-basis: 13vw;
  flex-grow: 1;
  padding: 0;
  text-align: center;
  margin: 10px 0 0 10px;
  display: inline-block;
`;

const C = styled.span`
  font-family: 'Monsieur La Doulaise', cursive;
  font-size: 3rem;
  color: #191718;
`;

const T = C.extend`
  display: inline-block;
  transform: translateX(-40px) translateY(8px);
`;

function MegaLogo() {
  return (
    <MegaLogoBase>
      <C>C</C>
      <T>T</T>
    </MegaLogoBase>
  );
}

MegaLogo.propTypes = {};

export default MegaLogo;
