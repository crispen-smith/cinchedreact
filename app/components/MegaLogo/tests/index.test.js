import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import MegaLogo from '../index';

describe('<MegaLogo />', () => {
  it('Renders a MegaLogoBase', () => {
    const tree = renderer.create(<MegaLogo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders a C', () => {
    const RenderedComponent = shallow(<MegaLogo />);
    expect(RenderedComponent.contains('C')).toEqual(true);
  });
  it('Renders a T', () => {
    const RenderedComponent = shallow(<MegaLogo />);
    expect(RenderedComponent.contains('T')).toEqual(true);
  });
});
