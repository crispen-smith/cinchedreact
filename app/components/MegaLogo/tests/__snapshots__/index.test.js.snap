// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MegaLogo /> Renders a MegaLogoBase 1`] = `
<div
  className="MegaLogo__MegaLogoBase-gVmnHe kfezCB"
>
  <span
    className="MegaLogo__C-hBLyQD CnhnX"
  >
    C
  </span>
  <span
    className="MegaLogo__C-hBLyQD iPmZWc"
  >
    T
  </span>
</div>
`;
